### PR TITLE
Automatic mysql 4byte detection

### PR DESCRIPTION
**Pull request by nickvergessen**
Downstream of missing commits from 17978 and 27441

### CODE REVIEW

File: lib/private/Repair/Collation.php [hunk 1]
```diff
@@ -107,6 +107,9 @@ public function run(IOutput $output) {
 		if (empty($tables)) {
 			$output->info('All tables already have the correct collation -> nothing to do');
 		}
+		if (empty($tables)) {
+			$output->info('All tables already have the correct collation -> nothing to do');
+		}
```
Comment (MorrisJobke) [inline lib/private/Repair/Collation.php]: I guess this is a merge conflict mishap 😉 - let me remove it

**Review by MorrisJobke — APPROVED** [review]
Tested and works 👍  (also the code makes sense)

**Comment by MorrisJobke** [comment]
@rullzer @blizzz @icewind1991 please review this one

File: config/config.sample.php [hunk 2]
```diff
@@ -1234,16 +1234,23 @@
...
 
 /**
- * If this setting is set to true MySQL can handle 4 byte characters instead of
- * 3 byte characters
- *
+ * During setup, if requirements are met (see below), this setting is set to true
+ * and MySQL can handle 4 byte characters instead of 3 byte characters.
+ *
+ * If you want to convert an existing 3-byte setup into a 4-byte setup please 
+ * set the parameters in MySQL as mentioned below run the migration command:
```
Comment (ChristophWurst) [inline config/config.sample.php]: "… set the parameters in MySQL as mentioned below **and** run the migration command: …"?

**Review by ChristophWurst — APPROVED** [review]
Changes make sense and code looks good. Just found a possible error in the config description.

File: config/config.sample.php (hunk 2 above)
Comment (nickvergessen) [inline config/config.sample.php]: fixed

**Review by LukasReschke — APPROVED** [review]
Makes sense 👍

**Pull request merged**